sam-react-app/coreconcept/src/moviesByGenre/MoviesByGenre.tsx

```diff
@@ -0,0 +1,37 @@
+import React, {Component, ChangeEvent} from "react";
+
+class MoviesByGenre extends Component {
+    state = {
+        genre: ["All", "Comedy", "Horror", "Crime", "Adventure", "Action"],
+        selectedGenre: 'All',
+        moviesMap: new Map([
+            ['Kill bill', ["Crime", "Adventure", "Horror", "All"]],
+            ["Inception", ["Action", "Adventure", "All"]],
+            ["Avenger Infinity War", ["Action", "All"]],
+            ["Pulp Fiction", ["Action", "Adventure", "All"]],
+            ["War Lord", ["Comedy", "Crime", "All"]],
+        ]),
+    };
+    handleOptionChange = (event: ChangeEvent<HTMLSelectElement>) => {
```

Comment:
This should come as a prop along with others:

> The component should accept three properties:
A list of genre names to display. Use the incoming list to render genre buttons.
A name of currently selected genre. Use the name to identify which button to highlight.
A "onSelect" callback property. Call the callback function when the user clicks on any genre button. Pass respective genre name to the callback arguments.

Reply:
done